### ISSUE TITLE
Depth peeling not functioning with xvfb

### ISSUE THREAD

**Issue by dyvasey** [pyvista/pyvista#2592]
**Describe the bug, what's wrong, and what you expect:**

I frequently use Pyvista to plot multiple 2D meshes on top of each other that are partially transparent. I find that the transparency only renders correctly when depth peeling is enabled. However, I recently needed to do some plotting on a remote machine (using xvfb) and find that transparent meshes will not render at all if depth peeling is enabled.

I expect when using xvfb in conjunction with depth peeling on a headless machines for transparency/opacity to render correctly. I provide an example code below using the planefile example.

-----

**To Reproduce**

Please include a code snippet to reproduce the bug in the block below:

```py
"""
Test for opacity on headless machine
"""

import pyvista as pv
from pyvista import examples

pv.start_xvfb()

mesh = pv.read(examples.planefile)

plotter = pv.Plotter(off_screen=True)
plotter.enable_depth_peeling()    

plotter.add_mesh(mesh, opacity=0.5)

plotter.show(screenshot='test.png')

```

**Screenshots**
Resulting screenshot using the code above:
![test_depth_peeling](https://user-images.githubusercontent.com/78242242/168188791-641f4ec2-53a5-42b3-b260-bfa75da49901.png)

Screenshot with depth peeling disabled:
![test_nodepthpeeling](https://user-images.githubusercontent.com/78242242/168188839-226f8e20-3ba9-4ba2-96d3-e3f551889a54.png)

-----

**System Information:**
 Please run the following code wherever you are experiencing the bug and paste the output below. This report helps us track down bugs and it is critical to addressing your bug:

```py
# Get system info
import pyvista as pv
print(pv.Report())
```

```txt
--------------------------------------------------------------------------------
  Date: Thu May 12 17:37:55 2022 PDT

                OS : Linux
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
               RAM : 27.3 GiB
       Environment : Python
       File system : ext4
       GPU Details : error

  Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:59)
  [GCC 10.3.0]

           pyvista : 0.34.0
               vtk : 9.1.0
             numpy : 1.22.3
           imageio : 2.19.1
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.2
             PyQt5 : 5.12.3
           IPython : 7.33.0
             scipy : 1.8.0
              tqdm : 4.64.0

```

**Comment by artcytracy** [comment]
I am having this issue also but I am not using xvbf or a remote machine. The depth peeling worked fine for a few months on my team's project, on various Mac and Windows machines, but suddenly started failing for some users, including myself, the author of the code. It still works as expected for one user and I'm trying to figure out what configuration might be different for that person.

```
Date: Mon Aug 01 12:55:02 2022 PDT

                OS : Darwin
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro 560 OpenGL Engine
       GPU Version : 4.1 ATI-4.8.101

  Python 3.9.9 (main, Jul 19 2022, 16:35:55)  [Clang 13.0.0
  (clang-1300.0.27.3)]

           pyvista : 0.34.2
               vtk : 9.1.0
             numpy : 1.23.1
           imageio : 2.20.0
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.2
```

**Comment by akaszynski** [comment]
@banesullivan, I know that many render passes weren't working for you. Does any of this ring a bell?

**Comment by artcytracy** [comment]
I've also run a simple test on my personal desktop and it works fine. 

Code:
```
import pyvista as pv
plot = pv.Plotter(off_screen=False)
plot.add_mesh(pv.Sphere(), opacity=0.7)
plot.add_mesh(pv.Sphere(radius=2), opacity=0.3)
plot.enable_depth_peeling()
plot.show()
```
System info:
```  
Date: Mon Aug 01 16:30:35 2022 Pacific Daylight Time
                OS : Windows
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce GTX 1660 SUPER/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 512.95
  Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64
  bit (AMD64)]
           pyvista : 0.36.1
               vtk : 9.1.0
             numpy : 1.23.1
           imageio : 2.21.0
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.2
```

**Comment by lopsided** [comment]
I'm having the exact same problem with `mayavi`, I believe it is a problem with `vtk`/`Xvfb`. Is there any alternative to `Xvfb` for headless rendering?

**Comment by akaszynski** [comment]
> I'm having the exact same problem with `mayavi`, I believe it is a problem with `vtk`/`Xvfb`. Is there any alternative to `Xvfb` for headless rendering?

Yes, OSMesa, see https://github.com/pyvista/pyvista-wheels

**Comment by lopsided** [comment]
Thanks @akaszynski. I started again with a minimal conda environment (python=3.9, no `mayavi`). I installed `vtk` with `pip install https://github.com/pyvista/pyvista-wheels/raw/main/vtk-osmesa-9.1.0-cp39-cp39-linux_x86_64.whl` then installed `pyvista` with `pip install pyvista`. The OP example above (without the `pv.start_xvfb()` line) then failed with `ImportError: libOSMesa.so.8: cannot open shared object file: No such file or directory` but this was resolved with `apt-get install libosmesa6`.

The example now runs, both with and without `enable_depth_peeling()` -- great! Unfortunately with `enable_depth_peeling()` it still gives a blank image as per the OP. Therefore it doesn't look like an issue with `Xvfb` after all. Any other ideas?

**Comment by lopsided** [comment]
[Duplicating my 2 comments from https://github.com/enthought/mayavi/issues/1165]

TL;DR: `export GALLIUM_DRIVER=softpipe` fixes the problem.

The problem appears to be with the renderer. When using `Xvfb` you don't get the hardware support so even with a GPU on the headless machine you can't use it (AFAICT). That means you're using the software renderer `llvmpipe` by default. This can be changed to `swr` or `softpipe` (I found this from here: https://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D). I couldn't get `swr` to work but setting it to `softpipe` resolves the issue. Obviously don't expect the performance to be good using this option.

In an attempt to fix gpu support I tried running a full X server on my machine, which shows up as a process in `nvidia-smi` and gives me `OpenGL renderer string: Quadro RTX 4000/PCIe/SSE2` as output to `glxinfo | grep -i render`. However I _still_ need this fix to use depth peeling, so I'm not actually sure if I'm getting any gpu rendering here or not.

-- Update --
OK in a last attempt, I installed the `llvm-dev` package (https://docs.mesa3d.org/drivers/llvmpipe.html) and reinstalled all my conda packages. Nope, `GALLIUM_DRIVER=llvmpipe` still fails under `Xvfb`.

BUT (I'm sure coincidentally) now my gpu-aware Xorg server now renders correctly with any setting of `GALLIUM_DRIVER` (including the previously crashing `swr`) so I appear to have at least got gpu rendering working.

**Comment by dyvasey** [comment]
I can confirm that running `export GALLIUM_DRIVER=softpipe` prior to running a script using depth peeling as suggested above does solve this problem (and results in a performance hit). Thanks for this solution!
